packages/aws-cdk-lib/core/lib/removal-policys.ts

```diff
@@ -0,0 +1,171 @@
+import { IConstruct } from 'constructs';
+import { Aspects, IAspect } from './aspect';
+import { CfnResource } from './cfn-resource';
+import { RemovalPolicy } from './removal-policy';
+
+/**
+ * Properties for applying a removal policy
+ */
+export interface RemovalPolicyProps {
+  /**
+   * Apply the removal policy only to specific resource types.
+   * Can be a CloudFormation resource type string (e.g., 'AWS::S3::Bucket')
+   * or a class that extends CfnResource (e.g., TestBucketResource).
+   * @default - apply to all resources
+   */
+  readonly applyToResourceTypes?: (string | typeof CfnResource)[];
```

Comment:
This feature is very helpful for me (or us)!

But I'm not sure if we should use an Eum-Like (Union-Like) class for generation of public interfaces for other languages using a jsii translation instead of an union type, what do you think?

Reply:
After much deliberation, I decided on string[] because I thought it would work well enough. 

```ts
RemovalPolicies.of(parent).retain({
  applyToResourceTypes: [
    'AWS::DynamoDB::Table',
    bucket.cfnResourceType, // 'AWS::S3::Bucket'
  ],
});
```

Reply:
I'm OK! Could you please change the description and examples in this PR?

Reply:
fixed.

Reply:
We might want to fix that example as `cfnResourceType` appears to be a static method. (I often read the PR description to understand its feature, so correcting it would be more useful)

```ts
RemovalPolicies.of(scope).retain({
  applyToResourceTypes: [
       CfnBucket.cfnResourceType,
       CfnTable.cfnResourceType,
  ]
});
```

Reply:
I have made the correct Usage.